internal/sim/generators.go

```diff
@@ -43,6 +35,109 @@ func Flip(p float64) Generator[bool] {
+func Nat() Generator[int] {
+	choices := []Weighted[int]{
+		{100, Range(0, 10)},
+		{100, Range(10, 100)},
+		{100, Range(100, 1000)},
+		{100, Range(1000, 10_000)},
+		{100, Range(10_000, 100_000)},
+		{100, Range(100_000, 1_000_000)},
+		{100, Range(1_000_000, 1_000_000_000)},
+		{100, Range(1_000_000_000, 1_000_000_000_000)},
+		{100, Range(1_000_000_000_000, math.MaxInt)},
+		{1, OneOf(
+			math.MaxInt, math.MaxInt-1,
+			math.MaxInt32+1, math.MaxInt32, math.MaxInt32-1,
+			math.MaxInt16+1, math.MaxInt16, math.MaxInt16-1,
+			math.MaxInt8+1, math.MaxInt8, math.MaxInt8-1,
+		)},
+	}
+	return Weight(choices)
+}
+
+// Int returns a [Generator] that returns integers. Note that Int does not
+// return all integers equiprobably. Instead, it biases towards numbers closer
+// to zero and other pathological numbers that are more likely to induce bugs
+// (e.g., math.MaxInt, math.MinInt).
+func Int() Generator[int] {
+	choices := []Weighted[int]{
+		{100, Range(-10, 10)},
+		{50, Range(10, 100)},
+		{50, Range(-100, -10)},
+		{50, Range(100, 1000)},
+		{50, Range(-1000, -100)},
+		{50, Range(1000, 10_000)},
+		{50, Range(-10_000, -1000)},
+		{50, Range(10_000, 100_000)},
+		{50, Range(-100_000, -10_000)},
+		{50, Range(100_000, 1_000_000)},
+		{50, Range(-1_000_000, -100_000)},
+		{50, Range(1_000_000, 1_000_000_000)},
+		{50, Range(-1_000_000_000, -1_000_000)},
+		{50, Range(1_000_000_000, 1_000_000_000_000)},
+		{50, Range(-1_000_000_000_000, -1_000_000_000)},
+		{50, Range(1_000_000_000_000, math.MaxInt)},
+		{50, Range(math.MinInt, -1_000_000_000_000)},
+		{1, OneOf(
+			math.MaxInt, math.MaxInt-1,
+			math.MinInt, math.MinInt+1,
+			math.MaxInt32+1, math.MaxInt32, math.MaxInt32-1,
+			math.MinInt32-1, math.MinInt32, math.MinInt32+1,
+			math.MaxInt16+1, math.MaxInt16, math.MaxInt16-1,
+			math.MinInt16-1, math.MinInt16, math.MinInt16+1,
+			math.MaxInt8+1, math.MaxInt8, math.MaxInt8-1,
+			math.MinInt8-1, math.MinInt8, math.MinInt8+1,
+		)},
+	}
+	return Weight(choices)
+}
+
+// Float64 returns a [Generator] that returns 64-bit floats. Note that Float64
+// does not return all floats equiprobably. Instead, it biases towards numbers
+// closer to zero and other pathological numbers that are more likely to induce
+// bugs (e.g., NaN, infinity, -infinity, -0).
+func Float64() Generator[float64] {
+	pathologies := OneOf(
+		math.NaN(),           // NaN
+		math.Inf(1),          // infinity
+		math.Inf(-1),         // -infinity
+		math.Copysign(0, -1), // -0
+	)
+	return generatorFunc[float64](func(r *rand.Rand) float64 {
+		if r.Intn(1000) == 0 {
+			// 0.1% of the time, return a pathological float.
+			return pathologies.Generate(r)
+		}
+		const stdev = 1e6
+		return r.NormFloat64() * stdev
+	})
+}
+
+// Rune returns a [Generator] that returns runes equiprobably.
+func Rune() Generator[rune] {
+	return generatorFunc[rune](func(r *rand.Rand) rune {
+		// Note that rune is an alias for int32.
+		return rune(r.Intn(math.MaxInt32 + 1))
```

Comment:
This random rune thing would break our code left and right! 🤣

Reply:
:rofl: